automated/android/noninteractive-tradefed/xts_module_sharding.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import os
+import sys
+from lxml import etree
+
+
+def get_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(description="XTS module static sharding parser")
+    parser.add_argument(
+        "-p",
+        "--test-path",
+        required=True,
+        # TODO: check the possibility to split cts plan/config.
+        choices=["android-vts"],
+        help="Path to tradefed package top directory.",
+    )
+    parser.add_argument(
+        "-t", "--test-params", required=True, help="Tradefed shell test parameters."
+    )
+    parser.add_argument(
+        "-n",
+        "--shard-number",
+        required=True,
+        help="The number of shards of a test module.",
+    )
+    parser.add_argument(
+        "-i",
+        "--shard-index",
+        required=True,
+        help="The index of the test module shard to run.",
+    )
+    parser.add_argument(
+        "-v", "--verbose", action="store_true", default=False, help="Be verbose."
+    )
+
+    return parser
+
+
+def shards(lst: list, n: int) -> list:
+    "Yield successive n-sized list shards from a given list."
+    for i in range(0, len(lst), n):
+        yield lst[i : i + n]
+
+
+def update_config(config: str, shard_number: int, shard_index: int) -> None:
+    """
+    Update module config to keep the tests_number//shard_number sized
+    shard #shard_index only.
+    """
+    tree = etree.parse(config)
+    root = tree.getroot()
+
+    # vts test modules like ltp and kselftest modules contain
```

Comment:
splitting cts test config is a different story. I looked into the big cts test module `testcases/CtsLibcoreTestCases/CtsLibcoreTestCases.config`, I don't see individual tests in the config.

```
    <test class="com.android.compatibility.testtype.LibcoreTest" >
        <option name="package" value="android.libcore.cts" />
        <option name="instrumentation-arg" key="filter"
                value="com.android.cts.core.runner.ExpectationBasedFilter" />
        <option name="core-expectation" value="/knownfailures.txt" />
        <option name="virtual-device-core-expectation" value="/virtualdeviceknownfailures.txt" />
        <option name="runtime-hint" value="45m"/>
        <!-- 20x default timeout of 600sec -->
        <option name="shell-timeout" value="12000000"/>
        <option name="hidden-api-checks" value="false"/>
        <option name="device-listeners" value="org.conscrypt.ConscryptInstrumentationListener" />
        <option name="instrumentation-arg" key="conscrypt_sslsocket_implementation" value="engine" />
    </test>
```

More investigation is required to split cts test module. Not in this MR.